### ISSUE TITLE
`.dart` and `.dartServer` is stored in home folder on macOS

### ISSUE THREAD

**Issue by probablykasper** [dart-lang/sdk#49166]
On macOS, Dart stores data in the unconventional folders `~/.dart` and `~/.dartServer`. This contributes to a mess where app files are scattered everywhere, the home folder is cluttered, and backups get filled with unnecessary junk.

This should be fixed and moved to proper paths, like `~/Library/Application Support`, `~/Library/Caches` and `~/Library/Preferences`.

* Dart SDK version: 2.17.1 (stable) (Tue May 17 17:58:21 2022 +0000) on "macos_x64"
* macOS 10.15

**Comment by lrhn** [comment]
Not sure who uses `.dart`, but `.dartServer` sounds like the analysis server.

**Comment by probablykasper** [comment]
`.dart` seems to be for user-level settings. If `.dartServer` is for the analysis server, is there a different repo this should be reported in?

**Comment by bwilkerson** [comment]
Yes, `.dartServer` is from the analysis server, and this repo is the right place for reporting the issue.

There's a larger conversation happening about unifying all of our tools and cleaning up where we're putting temporary files. Unfortunately, I don't remember who's leading that effort.

**Comment by bwilkerson** [comment]
@mit-mit Do you remember where this was recently discussed?

Assigning p3 until I can figure out what the larger plan is.

**Comment by mit-mit** [comment]
I think there was some discussion about this for pub, @jonasfj ?

I think we might be able to create a package with FFI for determining the canonical folders for each host OS?

**Comment by jonasfj** [comment]
See: https://github.com/dart-lang/sdk/issues/41560#issuecomment-919071070

But that's really only for configuration.

There is XDG specification details for where to put cache folders. So if we want `.dartServer` to live somewhere else we should probably add a `applicationCacheHome()` to `cli_util` similar to how I added `applicationConfigHome` in https://github.com/dart-lang/cli_util/pull/66

**Comment by derolf** [comment]
In your `.zshrc`, add:

``
export ANALYZER_STATE_LOCATION_OVERRIDE=$HOME/Library/Caches/dart_server
```

**Comment by bartekpacia** [comment]
related https://github.com/dart-lang/sdk/issues/42813
